### ISSUE TITLE
Problems with low accuracy

### ISSUE THREAD

**Issue by zjyLibra** [liyu10000/DeepLDA#3]
I took your code after debugging and it can run smoothly, but the accuracy rate of the run is very low, only 20 to 30%. I tried many batchsizes, and the result is still the same after many epochs. It can't reach you said the expected effect. When I remove the lda part and use CCE to calculate loss and accuracy, the effect is good and it can reach more than 80. This problem has troubled me for a long time. I don’t know if you have updated your code, or similar situations have occurred. ? Looking forward to your reply.

**Comment by liyu10000** [comment]
Hi,
Do you mind to tell me which dataset you are using? DeepLDA is weak when task is hard. Also the number of classes cannot be high, e.g. 20 or higher.

**Comment by zjyLibra** [comment]
I'm very glad to receive your reply. The data set I used is your cifar10, 32 * 32, train.py. I didn't make any changes. Lda.py may have some changes due to the version problem under the window system, but I can't get the expected effect. I'm very confused
![F07761E463AE7249B644FDF31B4C42BE](https://user-images.githubusercontent.com/66671420/119075077-c4e61700-ba22-11eb-8eb4-99b92b9d70d7.png)
![FB9EA0BDF5FD876046A7360A030AC2CB](https://user-images.githubusercontent.com/66671420/119075084-c8799e00-ba22-11eb-95db-d7f2486a1c17.png)
The first picture is the result of removing the LDA part, and the second picture is the result of keeping the LDA part.

**Comment by zjyLibra** [comment]
Thank you for your reply, I have carried out debugging again, the recurrence is successful, thank you very much~

**Comment by zjyLibra** [comment]
But I found a question, why did the training have a high accuracy rate at the beginning, and the accuracy rate does not change much in the subsequent training process, have you encountered this situation?

**Comment by liyu10000** [comment]
Yeah the training with LDA is usually fast because it optimizes the objective directly. It however won't improve much once reached a good state.

**Comment by webzerg** [comment]
@zjyLibra can you share what went wrong on the low accuracy and how you debugged it to get the high accuracy? Thanks
